preloader.go

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ * Copyright 2014 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpc
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// PreparedMsg is responsible for creating a Marshalled and Compressed object
+type PreparedMsg struct {
+	// Struct for preparing msg before sending them
+	encodedData []byte
+	compredData []byte
```

Comment:
It is not used in any of the use cases right now, but is computed as an intermediate step when Encode is invoked. 

It is possible there is some implementation change to server / client which might want to use it later.

Reply:
This is a not exported field. So if we need it in the future because of an implementation change, we can add it later. For now, I don't think it is necessary.

Reply:
Makes sense. Fixed.